### PR TITLE
chore: unify-binaries-reading-and-download

### PR DESCRIPTION
**Pull request by Misieq01**

### CODE REVIEW

**Comment by Misieq01** [comment]
NOT READY FOR MERGE

File: src-tauri/src/binary_resolver.rs [hunk 1]
```diff
@@ -1,516 +0,0 @@
...
-    ) -> Result<VersionAsset, Error> {
-        todo!()
-    }
-}
-
-pub struct GithubReleasesAdapter {
-    pub repo: String,
-    pub owner: String,
-    pub semver: Option<VersionReq>,
-    pub version_pre_filter: Option<Regex>,
```
Comment (stringhandler) [inline src-tauri/src/binary_resolver.rs:78]: These fields are missing in your new file

File: src-tauri/src/binaries/adapter_github.rs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        info!(target: BINARY_RESOLVER_LOG_TARGET, "Looking for platform with suffix: {}", name_suffix);
+
+        let reg = Regex::new(name_suffix).unwrap();
+
+        let platform = version
+            .assets
+            .iter()
+            .find(|a| {
+                if let Some(ref specific) = self.specific_name {
```
Comment (stringhandler) [inline src-tauri/src/binaries/adapter_github.rs:96]: Please check the latest logic for this

File: src-tauri/src/binary_resolver.rs (hunk 1 above)
Comment (Misieq01) [reply src-tauri/src/binary_resolver.rs:78]: Hey, they are included in binary_manager

File: src-tauri/src/binaries/binaries_manager.rs [hunk 3]
```diff
@@ -0,0 +1,416 @@
+use std::{collections::HashMap, path::PathBuf, str::FromStr};
+
+use semver::{Version, VersionReq};
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    binaries::binaries_resolver::BINARY_RESOLVER_LOG_TARGET,
```
Comment (stringhandler) [inline src-tauri/src/binaries/binaries_manager.rs:7]: Why not have the log target here?

File: src-tauri/src/binaries/adapter_github.rs (hunk 2 above)
Comment (Misieq01) [reply src-tauri/src/binaries/adapter_github.rs:96]: From what I see it's the same as on main branch

File: src-tauri/src/binaries/binaries_manager.rs (hunk 3 above)
Comment (Misieq01) [reply src-tauri/src/binaries/binaries_manager.rs:7]: Good point

**Comment by Misieq01** [comment]
From what I see no changes of features are missing from original binary_resolver

**Comment by stringhandler** [comment]
lol sorry for taking so long

**Pull request merged**